src/modules/logger/log_writer_file.cpp

```diff
@@ -247,10 +365,34 @@ void LogWriterFile::run()
 				LogFileBuffer &buffer = _buffers[i];
 				size_t available = buffer.get_read_ptr(&read_ptr, &is_part);
 
+#if defined(PX4_CRYPTO)
+				// Split into 64-byte chunks, so it is good for encrypting in pieces
+				available = (available / 64) * 64;
+#endif
+
 				/* if sufficient data available or partial read or terminating, write data */
 				if (available >= min_available[i] || is_part || (!buffer._should_run && available > 0)) {
 					pthread_mutex_unlock(&_mtx);
 
+#if defined(PX4_CRYPTO)
+					/* This makes the following assumtions:
+					 * - the chipher size is always the
+					     same as the input size
+					 * - the encryption can be done in
+					     place. This is always taken care
```

Comment:
I'm wondering if the API should always be in-place (remove the output buffer arg), as we're generally working with limited memory.

Reply:
I think this is a valid restriction and a good suggestion. If you must have an algorithm that can't do it in-place, you just need to handle it in the lower level.

Reply:
There we actually need to have separate pointers; in some cases (e.g. RSA with OAEP padding), the output can be of different size than the input; also sometimes you really want to also have a copy of the data, and it can be done at the same (no need for another copy). So I'd suggest keeping this  as is, but just require that all algorithms must work also in-place. Not sure how to enforce that though

Reply:
Ok. Then we just have to explicitly document the API in detail (i.e. `cipher_size` is an input & output param, and how a caller knows when the output size differs), otherwise we'll have subtle failures when switching between algorithms.

> Not sure how to enforce that though

It could be done with a unit test.